fdbserver/EncryptKeyProxy.actor.cpp

```diff
@@ -654,6 +680,66 @@ bool isBlobMetadataEligibleForRefresh(const BlobMetadataDetailsRef& blobMetadata
 	return nextRefreshCycleTS > blobMetadata.expireAt || nextRefreshCycleTS > blobMetadata.refreshAt;
 }
 
+ACTOR Future<KMSHealthStatus> getHealthStatusImpl(Reference<EncryptKeyProxyData> ekpProxyData,
+                                                  KmsConnectorInterface kmsConnectorInf) {
+	state UID debugId = deterministicRandom()->randomUniqueID();
+	if (DEBUG_ENCRYPT_KEY_PROXY) {
+		TraceEvent("KMSHealthCheckStart", ekpProxyData->myId);
+	}
+
+	// Health check will try to fetch the encryption details for the system key
+	try {
+		KmsConnLookupEKsByDomainIdsReq req;
+		req.debugId = debugId;
+		req.encryptDomainIds.push_back(SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID);
+		++ekpProxyData->numHealthCheckRequests;
+		KmsConnLookupEKsByDomainIdsRep rep = wait(timeoutError(kmsConnectorInf.ekLookupByDomainIds.getReply(req),
+		                                                       FLOW_KNOBS->EKP_HEALTH_CHECK_REQUEST_TIMEOUT));
+		if (rep.cipherKeyDetails.size() < 1) {
+			TraceEvent(SevWarn, "KMSHealthCheckResponseEmpty");
+			throw encrypt_keys_fetch_failed();
```

Comment:
nit: `error_code_encrypt_key_not_found` needs to be thrown as response was received and required key was not found?

Reply:
@sfc-gh-nwijetunga this is `marked` as resolved, but I thing the code is still the same? Maybe PR is not updated?